src/main/java/org/broadinstitute/hellbender/tools/spark/bwa/BwaSparkEngine.java

```diff
@@ -63,19 +63,102 @@ public BwaSparkEngine(final int numThreads, final int fixedChunkSize, final Stri
         this.referenceFileName = referenceFileName;
     }
 
+    /**
+     * Aligns the reads using bwa. The resulting reads are the same as the input reads but with updated alignment information.
+     */
     public JavaRDD<GATKRead> alignWithBWA(final JavaSparkContext ctx, final JavaRDD<GATKRead> unalignedReads, final SAMFileHeader readsHeader) {
         //Note: SparkContext is not serializable so we don't store it in the engine and set this property here. Setting it multiple times is fine.
         // ensure reads in a pair fall in the same partition (input split), so they are processed together
         ctx.hadoopConfiguration().setBoolean(BAMInputFormat.KEEP_PAIRED_READS_TOGETHER_PROPERTY, true);
 
-        final JavaRDD<Tuple2<ShortRead, ShortRead>> shortReadPairs = convertToUnalignedReadPairs(unalignedReads);
-        final JavaRDD<String> samLines = align(shortReadPairs);
+        final JavaRDD<Pair<BwaHelperShortRead, BwaHelperShortRead>> shortReadPairs = convertToUnalignedReadPairs(unalignedReads);
+        final JavaRDD<Pair<String, GATKRead>> samLines = align(shortReadPairs);
         final SAMLineParser samLineParser = new SAMLineParser(new DefaultSAMRecordFactory(), ValidationStringency.SILENT, readsHeader, null, null);
         final Broadcast<SAMLineParser> samLineParserBroadcast = ctx.broadcast(samLineParser);
-        return samLines.map(r -> new SAMRecordToGATKReadAdapter(samLineParserBroadcast.getValue().parseLine(r)));
+        return samLines.map(p -> {
+            final GATKRead originalRead = p.getRight();
+            final GATKRead alignedRead = new SAMRecordToGATKReadAdapter(samLineParserBroadcast.getValue().parseLine(p.getLeft()));
+
+            //we're going to just modify the original read
+            copyPositionalInfo(alignedRead, originalRead);
+            copyAlignmentRelatedTags(alignedRead, originalRead);
+            copyFlags(alignedRead, originalRead);
+
+            //we need to copy the bases and quals too because bwa may have reversed complemented them
+            originalRead.setBases(alignedRead.getBases());
+            originalRead.setBaseQualities(alignedRead.getBaseQualities());
+            return originalRead;
+        });
+    }
+
+    private static void copyPositionalInfo(final GATKRead from, final GATKRead to) {
+        if (from.isUnmapped()) {
+            to.setIsUnmapped();
```

Comment:
@lh3 Could you tell us whether BWA assigns nominal positions to unmapped reads with mapped mates?